modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java

```diff
@@ -174,34 +150,7 @@ public BType getType() {
 
     @Override
     public BValue copy() {
-        BMap map = BTypes.typeMap.getEmptyValue();
-        for (int i = 0; i < size; i++) {
-            BValue value = values[i].getValue();
-            map.put(values[i].getKey(), value == null ? null : value.copy());
-        }
-        return map;
-    }
-    
-    private class MapEntry<K, V> {
-        private final K key;
-        private V value;
-
-        MapEntry(K key, V value) {
-            this.key = key;
-            this.value = value;
-        }
-
-        K getKey() {
-            return key;
-        }
-
-        V getValue() {
-            return value;
-        }
-
-        void setValue(V value) {
-            this.value = value;
-        }
+        return new BMap<>(new LinkedHashMap(vals));
```

Comment:
I think this will not deep-copy the content in the `vals` HashMap (it will only shallow-copy). Need to explicitly copy the map.